Incorrect compiled signature of assignmentWithOffset in java API

https://github.com/akka/reactive-kafka/blob/9fbb88aa8995ca97d63d35c9fc23c0f1435faa4b/core/src/main/scala/akka/kafka/ConsumerSettings.scala#L84-L88

is meant to be used from Java, however, in the compiled bytecode the signature comes out as:
public static ManualSubscription assignmentWithOffset(Map<TopicPartition, Object> var0)

leading to compilation errors when called in java with a Map<TopicPartition, Long> argument:

> no suitable method found for assignmentWithOffset(java.util.Map<org.apache.kafka.common.TopicPartition,java.lang.Long>)
[ERROR] method akka.kafka.Subscriptions.assignmentWithOffset(java.util.Map<org.apache.kafka.common.TopicPartition,java.lang.Object>) is not applicable
[ERROR] (argument mismatch; java.util.Map<org.apache.kafka.common.TopicPartition,java.lang.Long> cannot be converted to java.util.Map<org.apache.kafka.common.TopicPartition,java.lang.Object>)
[ERROR] method akka.kafka.Subscriptions.assignmentWithOffset(scala.collection.Seq<scala.Tuple2<org.apache.kafka.common.TopicPartition,java.lang.Object>>) is not applicable
[ERROR] (argument mismatch; java.util.Map<org.apache.kafka.common.TopicPartition,java.lang.Long> cannot be converted to scala.collection.Seq<scala.Tuple2<org.apache.kafka.common.TopicPartition,java.lang.Object>>)
[ERROR] method akka.kafka.Subscriptions.assignmentWithOffset(scala.collection.immutable.Map<org.apache.kafka.common.TopicPartition,java.lang.Object>) is not applicable
[ERROR] (argument mismatch; java.util.Map<org.apache.kafka.common.TopicPartition,java.lang.Long> cannot be converted to scala.collection.immutable.Map<org.apache.kafka.common.TopicPartition,java.lang.Object>)

--- a/core/src/main/scala/akka/kafka/ConsumerSettings.scala
+++ b/core/src/main/scala/akka/kafka/ConsumerSettings.scala
@@ -85,7 +85,7 @@ object Subscriptions {
    * Manually assign given topics and partitions with offsets
    * JAVA API
    */
-  def assignmentWithOffset(tps: java.util.Map[TopicPartition, Long]): ManualSubscription = assignmentWithOffset(tps.asScala.toMap)
+  def assignmentWithOffset(tps: java.util.Map[TopicPartition, java.lang.Long]): ManualSubscription = assignmentWithOffset(tps.asScala.toMap.asInstanceOf[Map[TopicPartition, Long]])
 
   /**
    * Manually assign given topics and partitions with offsets
@@ -106,7 +106,7 @@ object Subscriptions {
    * Manually assign given topics and partitions with offsets
    * JAVA API
    */
-  def assignmentOffsetsForTimes(tps: java.util.Map[TopicPartition, Long]): ManualSubscription = assignmentOffsetsForTimes(tps.asScala.toMap)
+  def assignmentOffsetsForTimes(tps: java.util.Map[TopicPartition, java.lang.Long]): ManualSubscription = assignmentOffsetsForTimes(tps.asScala.toMap.asInstanceOf[Map[TopicPartition, Long]])
 
   /**
    * Manually assign given topics and partitions with offsets

